pygin/pygin/src/py.object.hpp

```diff
@@ -245,12 +245,14 @@ namespace py
 		using value_type = object;
 		using pointer = value_type*;
 
-		explicit iterator(const object& Container, bool IsEnd);
+		iterator(); // end, also support default_initializable concept for input_iterator
+		explicit iterator(const object& Container);
 
 		[[nodiscard]] object operator*() const;
 		[[nodiscard]] const object* operator->() const;
 
 		iterator& operator++();
+		void operator++(int); // post increment is required for an input iterator though it can return void
```

Comment:
I experimented with C++ ranges and they want this post-increment operator and a default ctor even for basic use cases (`input_range` concept). I ended up not using C++ ranges in this PR (that chunk of code was eliminated at some point) but I decided to leave this commit because the original compiler messages were really cryptic and I spend a considerable amount of time figuring out what the compiler wants.

Returning `void` was suggested as a possible solution somewhere on internet :)